### PR TITLE
sql/mysql: support reverse drop fk

### PR DESCRIPTION
**Pull request by hedwigz**

### CODE REVIEW

File: sql/mysql/migrate.go [hunk 1]
```diff
@@ -377,7 +377,11 @@ func (s *state) alterTable(t *schema.Table, changes []schema.Change) error {
 			reverse.Comma().P("DROP FOREIGN KEY").Ident(change.F.Symbol)
 		case *schema.DropForeignKey:
 			b.P("DROP FOREIGN KEY").Ident(change.F.Symbol)
-			reversible = false
+			reverse.Comma().P("ADD")
+			if err := s.fks(reverse, change.F); err != nil {
+				errors = append(errors, err.Error())
+			}
+			reversible = true
```
Comment (a8m) [inline sql/mysql/migrate.go:384]:
```suggestion
```

Comment (a8m) [reply sql/mysql/migrate.go:384]: It’s the default. See variable above

File: sql/mysql/migrate.go [hunk 2]
```diff
@@ -377,7 +377,11 @@ func (s *state) alterTable(t *schema.Table, changes []schema.Change) error {
 			reverse.Comma().P("DROP FOREIGN KEY").Ident(change.F.Symbol)
 		case *schema.DropForeignKey:
 			b.P("DROP FOREIGN KEY").Ident(change.F.Symbol)
-			reversible = false
+			reverse.Comma().P("ADD")
```
Comment (a8m) [inline sql/mysql/migrate.go:380]: A nice approach to reduce duplications (also in drop index), is to:

reverse.Comma()
b, reverse = reverse, b
defer func() { b, reverse = reverse, b }()
falltrough to add version.

File: sql/mysql/migrate_test.go [hunk 3]
```diff
@@ -213,6 +213,48 @@ func TestPlanChanges(t *testing.T) {
...
+							{Name: "id", Type: &schema.ColumnType{Type: &schema.IntegerType{T: "bigint"}}},
+							{Name: "user_id",
+								Type: &schema.ColumnType{
+									Type: &schema.IntegerType{T: "bigint"},
+								},
+							},
+						},
+					}
+					fk := &schema.ForeignKey{Symbol: "user_id", Table: pets, OnUpdate: schema.NoAction, OnDelete: schema.Cascade, RefTable: users}
+					pets.ForeignKeys = []*schema.ForeignKey{fk}
```
Comment (a8m) [inline sql/mysql/migrate_test.go:237]: Please add columns and ref-columns to this examp

File: sql/mysql/migrate.go (hunk 2 above)
Comment (hedwigz) [reply sql/mysql/migrate.go:380]: `fallthrough` does not work in type switch

Comment (a8m) [reply sql/mysql/migrate.go:380]: right :| 

kk, I have another idea, but let's merge this one atm

**Review by a8m — APPROVED** [review]

**Pull request merged**